File: Nx3CS/bin/test/src/nx3/PBeamgroupFrameCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PBeamgroupFrameCalculator : global::haxe.lang.HxObject {
		public    PBeamgroupFrameCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PBeamgroupFrameCalculator(global::nx3.PBeamgroup beamgroup){
			unchecked {
				#line 12 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				global::nx3.PBeamgroupFrameCalculator.__hx_ctor_nx3_PBeamgroupFrameCalculator(this, beamgroup);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PBeamgroupFrameCalculator(global::nx3.PBeamgroupFrameCalculator __temp_me88, global::nx3.PBeamgroup beamgroup){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				__temp_me88.beamgroup = beamgroup;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return new global::nx3.PBeamgroupFrameCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return new global::nx3.PBeamgroupFrameCalculator(((global::nx3.PBeamgroup) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PBeamgroup beamgroup;
		
		public  global::Array<int> outerLevels;
		
		public  global::Array<int> innerLevels;
		
		public virtual   object getFrame(){
			unchecked {
				#line 22 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				this.calcLevelArrays();
				object frame = this.calcFramePrototype();
				return frame;
			}
			#line default
		}
		
		
		public virtual   object calcFramePrototype(){
			unchecked {
				#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				int count = this.innerLevels.length;
				object tips = this.calcTips();
				global::Array<global::nx3.EBeamflagType> beamflags = new global::nx3.PBeamflagCalculator(((global::nx3.PBeamgroup) (this.beamgroup) )).getBeamflags();
				#line 34 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				{
					#line 39 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					int __temp_odecl245 = ((int) (global::haxe.lang.Runtime.getField_f(tips, "leftTip", 1961201556, true)) );
					int __temp_odecl246 = ((int) (global::haxe.lang.Runtime.getField_f(tips, "rightTip", 1679810271, true)) );
					#line 34 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					return new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{223262885, 711309176, 1178007882}), new global::Array<object>(new object[]{this.innerLevels, beamflags, this.outerLevels}), new global::Array<int>(new int[]{68419327, 935535770, 998862565, 1235217770, 1408766533, 1979547770}), new global::Array<double>(new double[]{((double) (this.innerLevels[( count - 1 )]) ), ((double) (__temp_odecl246) ), ((double) (this.outerLevels[0]) ), ((double) (this.innerLevels[0]) ), ((double) (__temp_odecl245) ), ((double) (this.outerLevels[( count - 1 )]) )}));
				}
				
			}
			#line default
		}
		
		
		public virtual   global::Array<int> getTopLevels(){
			unchecked {
				#line 50 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				global::Array<int> levels = new global::Array<int>(new int[]{});
				{
					#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					int _g = 0;
					#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::Array<object> _g1 = this.beamgroup.pnotes;
					#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					while (( _g < _g1.length )){
						#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						global::nx3.PNote note = ((global::nx3.PNote) (_g1[_g]) );
						#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						 ++ _g;
						#line 51 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						levels.push(note.nnote.getTopLevel());
					}
					
				}
				
				#line 52 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return levels;
			}
			#line default
		}
		
		
		public virtual   global::Array<int> getBottomLevels(){
			unchecked {
				#line 57 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				global::Array<int> levels = new global::Array<int>(new int[]{});
				{
					#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					int _g = 0;
					#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::Array<object> _g1 = this.beamgroup.pnotes;
					#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					while (( _g < _g1.length )){
						#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						global::nx3.PNote note = ((global::nx3.PNote) (_g1[_g]) );
						#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						 ++ _g;
						#line 58 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						levels.push(note.nnote.getBottomLevel());
					}
					
				}
				
				#line 59 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return levels;
			}
			#line default
		}
		
		
		public virtual   void calcLevelArrays(){
			unchecked {
				#line 64 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				global::nx3.EDirectionUD _g = this.beamgroup.getDirection();
				#line 64 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				switch (global::Type.enumIndex(_g)){
					case 0:
					{
						#line 67 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.outerLevels = this.getTopLevels();
						this.innerLevels = this.getBottomLevels();
						#line 66 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						break;
					}
					
					
					case 1:
					{
						#line 70 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.outerLevels = this.getBottomLevels();
						this.innerLevels = this.getTopLevels();
						#line 69 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						break;
					}
					
					
				}
				
			}
			#line default
		}
		
		
		public virtual   object calcTips(){
			unchecked {
				#line 79 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				int stemLenght = 7;
				global::nx3.EDirectionUD direction = this.beamgroup.getDirection();
				#line 82 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				global::nx3.PBamegroupFrameTipCalculator calculator = new global::nx3.PBamegroupFrameTipCalculator(((global::Array<int>) (this.outerLevels) ), ((global::nx3.EDirectionUD) (direction) ));
				#line 85 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				object tips = calculator.getTips();
				#line 87 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				if (( direction == global::nx3.EDirectionUD.Up )) {
					#line 87 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::haxe.lang.Runtime.setField_f(tips, "leftTip", 1961201556, ((double) (( ((int) (global::haxe.lang.Runtime.getField_f(tips, "leftTip", 1961201556, true)) ) - ((int) (stemLenght) ) )) ));
				}
				 else {
					#line 87 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::haxe.lang.Runtime.setField_f(tips, "leftTip", 1961201556, ((double) (( ((int) (global::haxe.lang.Runtime.getField_f(tips, "leftTip", 1961201556, true)) ) + stemLenght )) ));
				}
				
				#line 88 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				if (( direction == global::nx3.EDirectionUD.Up )) {
					#line 88 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::haxe.lang.Runtime.setField_f(tips, "rightTip", 1679810271, ((double) (( ((int) (global::haxe.lang.Runtime.getField_f(tips, "rightTip", 1679810271, true)) ) - ((int) (stemLenght) ) )) ));
				}
				 else {
					#line 88 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					global::haxe.lang.Runtime.setField_f(tips, "rightTip", 1679810271, ((double) (( ((int) (global::haxe.lang.Runtime.getField_f(tips, "rightTip", 1679810271, true)) ) + stemLenght )) ));
				}
				
				#line 90 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return tips;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				switch (hash){
					case 223262885:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.innerLevels = ((global::Array<int>) (global::Array<object>.__hx_cast<int>(((global::Array) (@value) ))) );
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return @value;
					}
					
					
					case 1178007882:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.outerLevels = ((global::Array<int>) (global::Array<object>.__hx_cast<int>(((global::Array) (@value) ))) );
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return @value;
					}
					
					
					case 1104035696:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.beamgroup = ((global::nx3.PBeamgroup) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				switch (hash){
					case 1194361325:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calcTips") ), ((int) (1194361325) ))) );
					}
					
					
					case 509692617:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calcLevelArrays") ), ((int) (509692617) ))) );
					}
					
					
					case 1076621680:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getBottomLevels") ), ((int) (1076621680) ))) );
					}
					
					
					case 2078397294:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getTopLevels") ), ((int) (2078397294) ))) );
					}
					
					
					case 1426166570:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calcFramePrototype") ), ((int) (1426166570) ))) );
					}
					
					
					case 1717209015:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getFrame") ), ((int) (1717209015) ))) );
					}
					
					
					case 223262885:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.innerLevels;
					}
					
					
					case 1178007882:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.outerLevels;
					}
					
					
					case 1104035696:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.beamgroup;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				switch (hash){
					case 1194361325:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.calcTips();
					}
					
					
					case 509692617:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						this.calcLevelArrays();
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						break;
					}
					
					
					case 1076621680:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.getBottomLevels();
					}
					
					
					case 2078397294:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.getTopLevels();
					}
					
					
					case 1426166570:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.calcFramePrototype();
					}
					
					
					case 1717209015:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return this.getFrame();
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				return default(object);
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				baseArr.push("innerLevels");
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				baseArr.push("outerLevels");
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				baseArr.push("beamgroup");
				#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
				{
					#line 9 "F:\\nx3\\src\\nx3\\PBeamgroupFrameCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PBeamgroupFrameCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PBeamgroupFrameCalculator extends haxe.lang.HxObject
{
	public    PBeamgroupFrameCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PBeamgroupFrameCalculator(nx3.PBeamgroup beamgroup)
	{
		nx3.PBeamgroupFrameCalculator.__hx_ctor_nx3_PBeamgroupFrameCalculator(this, beamgroup);
	}
	
	
	public static   void __hx_ctor_nx3_PBeamgroupFrameCalculator(nx3.PBeamgroupFrameCalculator __temp_me105, nx3.PBeamgroup beamgroup)
	{
		__temp_me105.beamgroup = beamgroup;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PBeamgroupFrameCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PBeamgroupFrameCalculator(((nx3.PBeamgroup) (arr.__get(0)) ));
	}
	
	
	public  nx3.PBeamgroup beamgroup;
	
	public  haxe.root.Array<java.lang.Object> outerLevels;
	
	public  haxe.root.Array<java.lang.Object> innerLevels;
	
	public   java.lang.Object getFrame()
	{
		this.calcLevelArrays();
		java.lang.Object frame = this.calcFramePrototype();
		return frame;
	}
	
	
	public   java.lang.Object calcFramePrototype()
	{
		int count = this.innerLevels.length;
		java.lang.Object tips = this.calcTips();
		haxe.root.Array<nx3.EBeamflagType> beamflags = new nx3.PBeamflagCalculator(((nx3.PBeamgroup) (this.beamgroup) )).getBeamflags();
		{
			int __temp_odecl650 = ((int) (haxe.lang.Runtime.toInt(this.innerLevels.__get(0))) );
			int __temp_odecl651 = ((int) (haxe.lang.Runtime.toInt(this.outerLevels.__get(0))) );
			int __temp_odecl652 = ((int) (haxe.lang.Runtime.toInt(this.innerLevels.__get(( count - 1 )))) );
			int __temp_odecl653 = ((int) (haxe.lang.Runtime.toInt(this.outerLevels.__get(( count - 1 )))) );
			double __temp_odecl654 = ((double) (haxe.lang.Runtime.getField_f(tips, "leftTip", true)) );
			double __temp_odecl655 = ((double) (haxe.lang.Runtime.getField_f(tips, "rightTip", true)) );
			return new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"beamflags", "innerLevels", "outerLevels"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{beamflags, this.innerLevels, this.outerLevels}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"leftInnerY", "leftOuterY", "leftTipY", "rightInnerY", "rightOuterY", "rightTipY"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (__temp_odecl650) )) ), ((java.lang.Object) (((double) (__temp_odecl651) )) ), ((java.lang.Object) (__temp_odecl654) ), ((java.lang.Object) (((double) (__temp_odecl652) )) ), ((java.lang.Object) (((double) (__temp_odecl653) )) ), ((java.lang.Object) (__temp_odecl655) )}));
		}
		
	}
	
	
	public   haxe.root.Array<java.lang.Object> getTopLevels()
	{
		haxe.root.Array<java.lang.Object> levels = new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{});
		{
			int _g = 0;
			haxe.root.Array<nx3.PNote> _g1 = this.beamgroup.pnotes;
			while (( _g < _g1.length ))
			{
				nx3.PNote note = _g1.__get(_g);
				 ++ _g;
				levels.push(note.nnote.getTopLevel());
			}
			
		}
		
		return levels;
	}
	
	
	public   haxe.root.Array<java.lang.Object> getBottomLevels()
	{
		haxe.root.Array<java.lang.Object> levels = new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{});
		{
			int _g = 0;
			haxe.root.Array<nx3.PNote> _g1 = this.beamgroup.pnotes;
			while (( _g < _g1.length ))
			{
				nx3.PNote note = _g1.__get(_g);
				 ++ _g;
				levels.push(note.nnote.getBottomLevel());
			}
			
		}
		
		return levels;
	}
	
	
	public   void calcLevelArrays()
	{
		nx3.EDirectionUD _g = this.beamgroup.getDirection();
		switch (haxe.root.Type.enumIndex(_g))
		{
			case 0:
			{
				this.outerLevels = this.getTopLevels();
				this.innerLevels = this.getBottomLevels();
				break;
			}
			
			
			case 1:
			{
				this.outerLevels = this.getBottomLevels();
				this.innerLevels = this.getTopLevels();
				break;
			}
			
			
		}
		
	}
	
	
	public   java.lang.Object calcTips()
	{
		int stemLenght = 7;
		nx3.EDirectionUD direction = this.beamgroup.getDirection();
		nx3.PBamegroupFrameTipCalculator calculator = new nx3.PBamegroupFrameTipCalculator(((haxe.root.Array<java.lang.Object>) (this.outerLevels) ), ((nx3.EDirectionUD) (direction) ));
		java.lang.Object tips = calculator.getTips();
		if (( direction == nx3.EDirectionUD.Up )) 
		{
			haxe.lang.Runtime.setField_f(tips, "leftTip", ((double) (( ((double) (haxe.lang.Runtime.getField_f(tips, "leftTip", true)) ) - ((double) (stemLenght) ) )) ));
		}
		 else 
		{
			haxe.lang.Runtime.setField_f(tips, "leftTip", ((double) (( ((double) (haxe.lang.Runtime.getField_f(tips, "leftTip", true)) ) + stemLenght )) ));
		}
		
		if (( direction == nx3.EDirectionUD.Up )) 
		{
			haxe.lang.Runtime.setField_f(tips, "rightTip", ((double) (( ((double) (haxe.lang.Runtime.getField_f(tips, "rightTip", true)) ) - ((double) (stemLenght) ) )) ));
		}
		 else 
		{
			haxe.lang.Runtime.setField_f(tips, "rightTip", ((double) (( ((double) (haxe.lang.Runtime.getField_f(tips, "rightTip", true)) ) + stemLenght )) ));
		}
		
		return tips;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef656 = true;
			switch (field.hashCode())
			{
				case -795379867:
				{
					if (field.equals("innerLevels")) 
					{
						__temp_executeDef656 = false;
						this.innerLevels = ((haxe.root.Array<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1993942992:
				{
					if (field.equals("beamgroup")) 
					{
						__temp_executeDef656 = false;
						this.beamgroup = ((nx3.PBeamgroup) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1972394442:
				{
					if (field.equals("outerLevels")) 
					{
						__temp_executeDef656 = false;
						this.outerLevels = ((haxe.root.Array<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef656) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef657 = true;
			switch (field.hashCode())
			{
				case -180941011:
				{
					if (field.equals("calcTips")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calcTips"))) );
					}
					
					break;
				}
				
				
				case -1993942992:
				{
					if (field.equals("beamgroup")) 
					{
						__temp_executeDef657 = false;
						return this.beamgroup;
					}
					
					break;
				}
				
				
				case -1666874295:
				{
					if (field.equals("calcLevelArrays")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calcLevelArrays"))) );
					}
					
					break;
				}
				
				
				case 1972394442:
				{
					if (field.equals("outerLevels")) 
					{
						__temp_executeDef657 = false;
						return this.outerLevels;
					}
					
					break;
				}
				
				
				case -628661776:
				{
					if (field.equals("getBottomLevels")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getBottomLevels"))) );
					}
					
					break;
				}
				
				
				case -795379867:
				{
					if (field.equals("innerLevels")) 
					{
						__temp_executeDef657 = false;
						return this.innerLevels;
					}
					
					break;
				}
				
				
				case -448602002:
				{
					if (field.equals("getTopLevels")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getTopLevels"))) );
					}
					
					break;
				}
				
				
				case 1953517495:
				{
					if (field.equals("getFrame")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getFrame"))) );
					}
					
					break;
				}
				
				
				case 2073293802:
				{
					if (field.equals("calcFramePrototype")) 
					{
						__temp_executeDef657 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calcFramePrototype"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef657) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef658 = true;
			switch (field.hashCode())
			{
				case -180941011:
				{
					if (field.equals("calcTips")) 
					{
						__temp_executeDef658 = false;
						return this.calcTips();
					}
					
					break;
				}
				
				
				case 1953517495:
				{
					if (field.equals("getFrame")) 
					{
						__temp_executeDef658 = false;
						return this.getFrame();
					}
					
					break;
				}
				
				
				case -1666874295:
				{
					if (field.equals("calcLevelArrays")) 
					{
						__temp_executeDef658 = false;
						this.calcLevelArrays();
					}
					
					break;
				}
				
				
				case 2073293802:
				{
					if (field.equals("calcFramePrototype")) 
					{
						__temp_executeDef658 = false;
						return this.calcFramePrototype();
					}
					
					break;
				}
				
				
				case -628661776:
				{
					if (field.equals("getBottomLevels")) 
					{
						__temp_executeDef658 = false;
						return this.getBottomLevels();
					}
					
					break;
				}
				
				
				case -448602002:
				{
					if (field.equals("getTopLevels")) 
					{
						__temp_executeDef658 = false;
						return this.getTopLevels();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef658) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			
		}
		
		return null;
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("innerLevels");
		baseArr.push("outerLevels");
		baseArr.push("beamgroup");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


